tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Ibexa\Tests\Rest\Server\Output\ValueObjectVisitor;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\API\Repository\Values\Content\Location as ApiLocation;
+use eZ\Publish\Core\Repository\Values\Content\Content;
+use eZ\Publish\Core\Repository\Values\Content\Location;
+use eZ\Publish\Core\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentType;
+use EzSystems\EzPlatformRest\Tests\Output\ValueObjectVisitorBaseTest;
+use EzSystems\EzPlatformRest\Server\Output\ValueObjectVisitor;
+
+final class LocationTest extends ValueObjectVisitorBaseTest
+{
+    /** @var \eZ\Publish\API\Repository\LocationService|\PHPUnit\Framework\MockObject\MockObject */
+    private $locationServiceMock;
+
+    /** @var \eZ\Publish\API\Repository\ContentService|\PHPUnit\Framework\MockObject\MockObject */
+    private $contentServiceMock;
+
+    protected function setUp(): void
+    {
+        $this->locationServiceMock = $this->createMock(LocationService::class);
+        $this->contentServiceMock = $this->createMock(ContentService::class);
+
+        parent::setUp();
+    }
+
+    public function testVisitWithDifferentLocationToMainLocation(): string
+    {
+        $visitor = $this->getVisitor();
+        $generator = $this->getGenerator();
+
+        $generator->startDocument(null);
+
+        $location = new Location([
+            'id' => 55,
+            'path' => ['1', '25', '42'],
+            'priority' => 1,
+            'sortField' => ApiLocation::SORT_FIELD_DEPTH,
+            'sortOrder' => ApiLocation::SORT_ORDER_ASC,
+            'parentLocationId' => 42,
+            'contentInfo' => new ContentInfo([
+                'id' => $contentId = 7,
+                'mainLocationId' => $mainLocationId = 78,
```

Comment:
Don't make such inline assignments, please. It's not very readable. Assign it earlier.